experimental/ast/type.go

```diff
@@ -42,6 +43,23 @@ type TypeKind int8
 // This type is used in lieu of a putative Type interface type to avoid heap
 // allocations in functions that would return one of many different Type*
 // types.
+//
+// # Grammar
+//
+//	Type := TypePath | TypePrefixed | TypeGeneric
+//
+// Note that there is an ambiguity when parsing a Type followed by a Path:
+// "optional optional foo" could be parsed as:
+//
+//	TypePrefix{Optional, TypePrefix{Optional, TypePath("foo")}}
+//
+// However, this should be parsed as follows:
+//
+//	TypePrefix{Optional, TypePath("optional")}, TypePath("foo")
```

Comment:
Did you by chance mean this (since the idea is parsing a Path after a Type, not two Types in a row)?
```suggestion
//	TypePrefix{Optional, TypePath("optional")}, Path("foo")
```